_posts/2023-10-14-bypass-oidc-trust.md

```diff
@@ -0,0 +1,129 @@
+---
+layout: postv2
+font: serif
+title: "Bypassing Github branch protection with Environments"
+description: "GitHub Environments changes the claim token provided to OIDC Providers and removes the Branch name from the Subject, this allows a user to authenticate via any branch"
+date: 2023-10-14
+highlight: monokai
+image: /img/bypass-oidc-trust/cover.jpg
+author: Chris Howard
+tags: [Trust policy, GitHub, GitHub Environments, Security, OIDC]
+---
+
+# Bypassing GitHub branch protection with OIDC and Github Environments
+
+As part of the deploying to AWS with Github Actions is the idea of using an OIDC provider to allow for temporary credentials that can be assumed during the pipeline to enable secure access to AWS accounts.
+
+The documentation for which is located [here](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services#adding-the-identity-provider-to-aws) shows that you should configure your trust policy like so:
+
+```
+"Condition": {
+  "StringEquals": {
+    "token.actions.githubusercontent.com:aud": "sts.amazonaws.com",
+    "token.actions.githubusercontent.com:sub": "repo:octo-org/octo-repo:ref:refs/heads/octo-branch"
+  }
+}
+```
+
+This configuration ensures that the only way that anyone can assume the deployment role is if the subject (sub) matches the specific branch name `octo-branch` if a deployment attempts to assume the role outside of that branch it will be denied, combined with standard branch protections (no code directly to branch, must have been via a Pull Request and have approvals by other approved members) ensures that the only way code can be deployed is via the review and approval process.
```

Comment:
```suggestion
This configuration ensures that the only way that anyone can assume the deployment role is if the subject (sub) matches the specific branch name `octo-branch`.  If a deployment attempts to assume the role outside of that branch it will be denied, combined with standard branch protections (no code pushed directly to branch, must have been via a Pull Request and have approvals by other approved members) ensures that the only way code can be deployed is via the review and approval process.
```